src/editor/map/map_context.cpp

```diff
@@ -123,15 +123,18 @@ map_context::map_context(const map_context&& ctxt)
 	, game_classification_(ctxt.game_classification_)
 	, music_tracks_(ctxt.music_tracks_)
 {
-	for (size_t i = 0; i < ctxt.undo_stack_.size(); i++) {
-		undo_stack_.push_back(std::move(undo_stack_.front()));
-		undo_stack_.pop_front();
+	for(auto& a : ctxt.undo_stack_) {
+		undo_stack_.push_back(a->clone());
```

Comment:
Why would you need to clone if this is a _move_ constructor? Cloning is for copying. In a move constructor you can just steal ownership of the pointer.

Reply:
Remove that ctor, since it wasn't needed, but yes, that was bad. (My understanding of move semantics is very limited.)